### PR TITLE
Fixed compiler warnings.

### PR DESCRIPTION
**Pull request by Fraser999**

### CODE REVIEW

File: include/maidsafe/crux/detail/multiplexer.hpp [hunk 1]
```diff
@@ -137,7 +137,8 @@ class multiplexer : public std::enable_shared_from_this<multiplexer>
     using socket_map = std::map<endpoint_type, socket_base *>;
     socket_map sockets;
 
-    std::atomic<int> receive_calls;
+    using receive_calls_underlying_type = int;
+    std::atomic<receive_calls_underlying_type> receive_calls;
```
Comment (breese) [inline include/maidsafe/crux/detail/multiplexer.hpp]: What warning does this fix? Is it the `static_cast` below?

Comment (breese) [inline include/maidsafe/crux/detail/multiplexer.hpp]: Sorry, I read this diff before your accompaning email, where you explain it.

Comment (Fraser999) [inline include/maidsafe/crux/detail/multiplexer.hpp]: Yes.  It's unrelated to the `decltype` in the cast, even this emits a warning:

    std::int64_t i(0);
    static_cast<std::atomic<int>>(i);

Comment (Fraser999) [inline include/maidsafe/crux/detail/multiplexer.hpp]: And I answered your first note before seeing the second! :-)

Comment (breese) [inline include/maidsafe/crux/detail/multiplexer.hpp]: Per email discussion, please change the type to use
```
std::atomic<std::size_t> receive_calls;
```
and remove the `static_cast`s from the assertions below.

**Comment by benjaminbollen** [comment]
Thanks Bjorn

**Pull request merged**